### PR TITLE
criticality_score playbook

### PR DESCRIPTION
**Pull request by NeerajK007**
criticality_score playbook - prepare & publish-report function is WIP.
Launch, init, execute, is completed

### CODE REVIEW

File: playbooks/besman-criticality_score-playbook.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
+function __besman_init {
+	echo -e "\ninit - collecting required info to execute the playbook\n"
+	echo -e "check & install snap"
+	if ! [ -x "$(command -v snap)" ]; then
+		echo "installing snap..."
+		sudo apt update
```
Comment (asa1997) [inline playbooks/besman-criticality_score-playbook.sh:6]: @NeerajK007, make installations part of environment script. Only use playbook for execution

File: playbooks/besman-criticality_score-playbook.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	fi
+
+	echo -e "\ninit completed"
+}
+
+function __besman_execute {
+	echo -e "\nexecuting - the playbook"
+	echo -e "enter project GITHUB_URL of project to scan: (note : (-)without .git in URL)"
+	read project_github_url
+	criticality_score -depsdev-disable -format json $project_github_url
```
Comment (asa1997) [inline playbooks/besman-criticality_score-playbook.sh:57]: Move this to a steps file.

**Review by asa1997 — CHANGES_REQUESTED** [review]
@NeerajK007 Please review the comments.

Please update the playbook-metadata.json file

**Review by pramit-d — CHANGES_REQUESTED** [review]
@NeerajK007 Please add version for playbooks. File name should contain version number.
Use [sbom generator](https://github.com/Be-Secure/besecure-playbooks-store/blob/main/playbooks/besman-spdx-sbom-generator-0.0.1-playbook.sh) playbook as example.

File: playbooks/besman-criticality_score-0.0.1-playbook.sh [hunk 3]
```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+function __besman_init() {
+    __besman_echo_white "initialising"
+    export ASSESSMENT_TOOL_NAME="criticality_score"
+    export ASSESSMENT_TOOL_TYPE="criticality_score"
+    export ASSESSMENT_TOOL_VERSION="0.0.1"
```
Comment (asa1997) [inline playbooks/besman-criticality_score-0.0.1-playbook.sh:6]: @NeerajK007 tool version should be the actual version of the tool being used.

File: playbooks/besman-criticality_score-0.0.1-playbook.sh [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    export BESLAB_OWNER_TYPE="Organization"
+    export BESLAB_OWNER_NAME="Be-Secure"
+    export ASSESSMENT_TOOL_PLAYBOOK="besman-$ASSESSMENT_TOOL_TYPE-$ASSESSMENT_TOOL_VERSION-playbook.sh"
```
Comment (asa1997) [inline playbooks/besman-criticality_score-0.0.1-playbook.sh:9]: For playbook it can be 0.0.1 for now

**Review by asa1997 — CHANGES_REQUESTED** [review]
@NeerajK007 Please address the comments

File: playbooks/besman-criticality_score-0.0.1-playbook.sh [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+    git add "$DETAILED_REPORT_PATH" "$OSAR_PATH"
+    git commit -m "Added osar and detailed report"
+    git push origin main
+}
+
+function __besman_cleanup() {
+    local var_array=("ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_PLAYBOOK" "ASSESSMENT_TOOL_VERSION" "OSAR_PATH" "CRITICALITY_SCORE_PATH" "DETAILED_REPORT_PATH")
+
+    :'for var in "${var_array[@]}"; do
```
Comment (asa1997) [inline playbooks/besman-criticality_score-0.0.1-playbook.sh:100]: @NeerajK007 This is throwing error,

image

File: playbooks/besman-criticality_score-0.0.1-playbook.sh [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    export ASSESSMENT_TOOL_TYPE="criticality_score"
+    export ASSESSMENT_TOOL_VERSION="2.0.3"
+    export BESLAB_OWNER_TYPE="Organization"
+    export BESLAB_OWNER_NAME="Be-Secure"
+    export ASSESSMENT_TOOL_PLAYBOOK="besman-$ASSESSMENT_TOOL_TYPE-0.0.1-playbook.sh"
+
+    local steps_file_name="besman-$ASSESSMENT_TOOL_NAME-0.0.1-steps.sh"
+    export BESMAN_STEPS_FILE_PATH="$BESMAN_PLAYBOOK_DIR/$steps_file_name"
+
+    local var_array=("BESMAN_ARTIFACT_TYPE" "BESMAN_ARTIFACT_NAME" "BESMAN_ARTIFACT_VERSION" "BESMAN_ARTIFACT_URL" "BESMAN_ENV_NAME" "BESMAN_ARTIFACT_DIR" "ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_VERSION" "ASSESSMENT_TOOL_PLAYBOOK" "BESMAN_ASSESSMENT_DATASTORE_DIR" "BESMAN_TOOL_PATH" "BESMAN_ASSESSMENT_DATASTORE_URL" "BESLAB_OWNER_TYPE" "BESLAB_OWNER_NAME")
```
Comment (asa1997) [inline playbooks/besman-criticality_score-0.0.1-playbook.sh:14]: Please change the variables - BESLAB_OWNER_TYPE -> BESMAN_LAB_TYPE; BESLAB_OWNER_NAME -> BESMAN_LAB_NAME

**Review by asa1997 — CHANGES_REQUESTED** [review]
@NeerajK007 Please address the comments

File: playbooks/besman-criticality_score-0.0.1-playbook.sh [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        dir_path="${!dir}"
+
+        if [[ ! -d $dir_path ]]; then
+            __besman_echo_red "Could not find $dir_path"
+            flag=true
+        fi
+
+    done
+
+    # [[ ! -f $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME ]] && __besman_echo_red "Could not find artifact @ $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME" && flag=true
```
Comment (asa1997) [inline playbooks/besman-criticality_score-0.0.1-playbook.sh:41]: Replace this commented code to check if criticality_score is installed or not, and give an appropriate message

Comment (NeerajK007) [reply playbooks/besman-criticality_score-0.0.1-playbook.sh:41]: done

**Review by asa1997 — APPROVED** [review]
@NeerajK007 LGTM

**Pull request merged**